src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,309 @@
+    /**
+     * Default color of the shadow on the button.
+     */
+    private static final Color CIRCLE_SHADOW_COLOR = Color.GRAY;
+    /**
+     * Default boolean state of the switch.
+     */
+    private static final boolean SWITCH_STATE = false;
+    //</editor-fold>
+    //<editor-fold desc="Constructor with default values" defaultstate="collapsed">
+    /**
+     * Constructor.
+     * The toggle switch is built by initializing a layout which includes the sizing of the background (a rectangle)
+     * and the sizing of the button (a circle), as well as their colors. The transition of the button is initialized by
+     * adding an animation which also entails a color change. The transition works by subtracting the diameter of the
+     * circle from the width of the rectangle. A listener and a mouse listener trigger the animation
+     * when the button (circle) is clicked on. The rectangle height variable is used for its arc width and arc height,
+     * which provides the rounded shape of the switch.
+     */
+    public ToggleSwitch() {
+        super();
+        this.switchStateBooleanProperty = new SimpleBooleanProperty(SWITCH_STATE);
+        this.switchBackground = new Rectangle(RECTANGLE_WIDTH_VALUE, RECTANGLE_HEIGHT_VALUE);
+        this.switchBackground.setArcWidth(RECTANGLE_HEIGHT_VALUE);
+        this.switchBackground.setArcHeight(RECTANGLE_HEIGHT_VALUE);
+        this.switchBackground.setLayoutX(RECTANGLE_POSITION_VALUE);
+        this.switchBackground.setFill(RECTANGLE_COLOR_OFF);
+        this.switchBackground.setStroke(RECTANGLE_OUTLINE_COLOR);
+        this.switchButton = new Circle(CIRCLE_RADIUS_VALUE);
+        this.switchButton.setCenterX(CIRCLE_POSITION_X_VALUE);
+        this.switchButton.setCenterY(CIRCLE_POSITION_Y_VALUE);
+        this.switchButton.setFill(CIRCLE_COLOR);
+        this.switchButton.setStroke(CIRCLE_OUTLINE_COLOR);
+        this.switchButton.setEffect(new DropShadow(CIRCLE_SHADOW_RADIUS, CIRCLE_SHADOW_COLOR));
+        this.switchAnimation = new TranslateTransition(Duration.seconds(DURATION_VALUE));
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation = new FillTransition(Duration.seconds(DURATION_VALUE));
+        this.switchTransition = new ParallelTransition(this.switchAnimation, this.fillAnimation);
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation.setShape(this.switchBackground);
+        this.getChildren().addAll(this.switchBackground, this.switchButton);
+        //Listener
+        this.switchStateBooleanProperty.addListener((observable, oldValue, newValue) -> {
+            boolean tmpIsOn = newValue.booleanValue();
+            this.switchAnimation.setToX(tmpIsOn ? (RECTANGLE_WIDTH_VALUE - (2 * CIRCLE_RADIUS_VALUE)) : 0);
+            this.fillAnimation.setFromValue(tmpIsOn ? RECTANGLE_COLOR_OFF : RECTANGLE_COLOR_ON);
+            this.fillAnimation.setToValue(tmpIsOn ? RECTANGLE_COLOR_ON : RECTANGLE_COLOR_OFF);
```

Comment:
Usage of the default values should be replaced with calls to the getters, e.g. this.switchBackground.getWidth() or this.rectangleColorOn (a non-final class variable).